lighthouse-core/fraggle-rock/config/filters.js

```diff
@@ -294,6 +294,8 @@ function filterConfigByExplicitFilters(config, filters) {
     baseAuditIds = getAuditIdsInCategories(config.categories, onlyCategories);
   } else if (onlyAudits) {
     baseAuditIds = new Set();
+  } else if (!Object.keys(config.categories || {}).length) {
```

Comment:
nit: having trouble reading this, can you pull out the `undefined` case as `config.categories && ...` / `!config.categories || ...` or whatever it is